build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -134,10 +141,11 @@ class AssetGraph {
       }
     }
 
-    deletes.addAll(allNodes
-        .where((n) => n is GeneratedAssetNode && n.needsUpdate)
-        .map((n) => n.id));
-    return deletes;
+    // Delete all the invalidated assets, then remove them from the graph. This
```

Comment:
this is unfortunate... We're getting some pretty tight coupling here. It is nicer than the previous version.

Did you mean to mention AssetWriter rather than AssetReader? I don't understand why a reader would be involved with a delete.

I don't have any specific alternatives to offer, but we can take another look later.

Reply:
Updated to `AssetWriter` (although technically `AssetReader` does also require things to be in the graph).

I definitely agree that this isn't ideal, but I couldn't come up with a better solution that doesn't require a much larger refactor which isn't worth going into at this time.

Reply:
After some more thought I'm less concerned about this than I was, I think this is likely to be fine for the foreseeable future.